src/transforms/pipelines/config.rs

```diff
@@ -41,62 +42,115 @@ impl Clone for PipelineConfig {
     }
 }
 
+#[async_trait::async_trait]
+#[typetag::serde(name = "pipeline")]
+impl TransformConfig for PipelineConfig {
+    async fn build(&self, ctx: &TransformContext) -> crate::Result<Transform> {
+        let condition = if let Some(config) = &self.filter {
+            Some(config.build(&ctx.enrichment_tables)?)
+        } else {
+            None
+        };
+
+        let mut transforms = Vec::new();
+        for config in &self.transforms {
+            let transform = match config.build(ctx).await? {
+                Transform::Function(transform) => Box::new(transform),
+                Transform::Synchronous(transform) => transform,
+                _ => panic!("non-sync transform in pipeline: {:?}", config),
+            };
+            transforms.push(transform);
+        }
+        Ok(Transform::Synchronous(Box::new(Pipeline {
+            condition,
+            transforms,
+        })))
+    }
+
+    fn input(&self) -> Input {
+        Input::all()
+    }
+
+    fn outputs(&self, _: &schema::Definition) -> Vec<Output> {
+        vec![Output::default(DataType::all())]
+    }
+
+    fn transform_type(&self) -> &'static str {
+        "pipeline"
+    }
```

Comment:
It'd be nice to avoid this boilerplate for transforms that are not meant to be directly user-configurable. In this case, we're really only implementing `TransformConfig` because we're ultimately returning this from an expansion, so it's likely this probably goes away when expansion goes away.